core/src/main/java/org/apache/calcite/sql/validate/TableIdentifierWithIDNamespace.java

```diff
@@ -105,12 +109,23 @@ public class TableIdentifierWithIDNamespace extends AbstractNamespace {
 
   private SqlValidatorNamespace resolveImpl(SqlTableIdentifierWithID id) {
     final SqlNameMatcher nameMatcher = validator.catalogReader.nameMatcher();
+
+    //NOTE: BasicSqlType defaults to non null
+    BasicSqlType int_typ = new BasicSqlType(RelDataTypeSystem.DEFAULT, SqlTypeName.BIGINT);
+    // NOTE: we don't use this index anywhere, so we just default it to a random value
+    int newIdx = 999;
```

Comment:
We should confirm this doesn't depend on state in the future, but this should be good for now.